File: Sum Root to Leaf Numbers.js
var sumNumbers = function(root) {
    let total = 0, sum = 0;
    helper(root, sum);
    return total;
    
    function helper(node, sum) {
        if (!node) 
return;
        sum = sum*10 + node.val;
        
        if (!node.left && !node.right) {
            total += sum;
        }
        helper(node.left, sum);
        helper(node.right, sum);
    }
};

File: Sum Root to Leaf Numbers.c
int sumNumbers(struct TreeNode* root) {
    return helper(root, 0);
}

int helper(struct TreeNode* root, int sum)
{
    if(root == NULL)
        return 0;
    if(root -> left == NULL && root -> right == NULL)
        return sum*10 + root -> val;
    return helper(root -> left,sum*10+root -> val) + helper(root -> right,sum*10+root -> val);
}
